shell/browser/api/electron_api_browser_view.h

```diff
@@ -7,11 +7,14 @@
 
 #include <memory>
 #include <string>
+#include <vector>
 
 #include "content/public/browser/web_contents_observer.h"
 #include "gin/handle.h"
 #include "gin/wrappable.h"
+#include "shell/browser/api/electron_api_web_contents.h"
```

Comment:
Can you extract `ExtendedWebContentsObserver` to a separate header, it will help clean out some dependencies like including `shell/browser/api/electron_api_web_contents.h` in other headers.